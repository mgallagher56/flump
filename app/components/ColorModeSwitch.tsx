import type { FC } from 'react';

import { useColorMode } from '@chakra-ui/react';

import FLPButton from './core/buttons/FLPButton';

const ColorModeSwitch: FC = () => {
  const { colorMode, toggleColorMode } = useColorMode();
  return (
    <FLPButton padding={0} variant="outline" onPress={toggleColorMode}>
      {colorMode === 'light' ? '🌚' : '☀️'}
    </FLPButton>
  );
};

export default ColorModeSwitch;
