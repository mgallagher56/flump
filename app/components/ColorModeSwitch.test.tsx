import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import ColorModeSwitch from './ColorModeSwitch';

vi.mock('@chakra-ui/react');

// mock flp button
vi.mock('./core/buttons/FLPButton', () => ({
  default: (props) => <button>{props.children}</button>
}));

describe('<ColorModeSwitch />', async () => {
  const chakra = await import('@chakra-ui/react');
  chakra.useColorMode = vi
    .fn()
    .mockReturnValueOnce({ ...chakra.useColorMode, colorMode: 'dark' })
    .mockReturnValueOnce({ ...chakra.useColorMode, colorMode: 'light' });
  it('should render as expected when colormode is dark', () => {
    const { baseElement } = render(<ColorModeSwitch />);
    expect(baseElement).toMatchSnapshot();
  });
  it('should render as expected when colormode is light', () => {
    const { baseElement } = render(<ColorModeSwitch />);
    expect(baseElement).toMatchSnapshot();
  });
});
