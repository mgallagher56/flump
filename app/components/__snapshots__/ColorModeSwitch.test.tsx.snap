// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ColorModeSwitch /> > should render as expected when colormode is dark 1`] = `
<body>
  <div>
    <button
      class="chakra-button css-p7mkev"
      id="pressable:pressableBaseButton"
      type="button"
    >
      ðŸŒš
    </button>
  </div>
</body>
`;

exports[`<ColorModeSwitch /> > should render as expected when colormode is light 1`] = `
<body>
  <div>
    <button
      class="chakra-button css-p7mkev"
      id="pressable:pressableBaseButton"
      type="button"
    >
      ðŸŒš
    </button>
  </div>
</body>
`;
